superset-frontend/packages/superset-ui-chart-controls/src/components/InfoTooltipWithTrigger.tsx

```
@@ -16,65 +16,101 @@
-    bsStyle ? `text-${bsStyle}` : ''
-  }`;
+  iconStyle,
+}: InfoTooltipWithTriggerProps) => {
+  const theme = useTheme();
+
+  const infoTooltipWithTriggerVariants = useMemo(
+    () => ({
+      info: { color: theme.colorIcon, icon: <InfoCircleOutlined /> },
+      question: { color: theme.colorIcon, icon: <QuestionCircleOutlined /> },
+      warning: { color: theme.colorWarning, icon: <WarningOutlined /> },
+      notice: { color: theme.colorWarning, icon: <ThunderboltOutlined /> },
+      error: { color: theme.colorError, icon: <CloseCircleOutlined /> },
+    }),
+    [theme],
+  );
+
+  const variant = type ? infoTooltipWithTriggerVariants[type] : null;
+
+  const iconCss = css`
+    color: ${variant?.color ?? theme.colorIcon};
+    font-size: ${themeObject.getFontSize(iconSize)};
+  `;
+
+  const handleKeyDown = (event: KeyboardEvent<HTMLSpanElement>) => {
+    if (onClick && (event.key === 'Enter' || event.key === ' ')) {
+      onClick();
+    }
+  };
+
   const iconEl = (
-    <i
+    <span
       role="button"
       aria-label={t('Show info tooltip')}
       tabIndex={0}
-      className={iconClass}
-      style={{ cursor: onClick ? 'pointer' : undefined, ...iconsStyle }}
+      className={className}
+      css={[
+        css`
+          display: inline-flex;
+          align-items: center;
+          line-height: 0;
+          vertical-align: middle;
+          ${onClick ? 'cursor: pointer;' : ''}
+        `,
+        iconCss,
+        iconStyle,
+      ]}
       onClick={onClick}
-      onKeyPress={
-        onClick &&
-        (event => {
-          if (event.key === 'Enter' || event.key === ' ') {
-            onClick();
-          }
-        })
-      }
-    />
+      onKeyDown={handleKeyDown}
+    >
+      {variant?.icon}
+    </span>
   );
+
   if (!tooltip) {
     return iconEl;
   }
+
   return (
     <Tooltip
-      id={`${kebabCase(label)}-tooltip`}
+      id={`${kebabCase(label) || Math.floor(Math.random() * 10000)}-tooltip`}
```

I don't think we should do this. If it is necessary to default to random id, there should be a useId hook for use.

---

I'll keep it in mind, thanks!

---

I think we should still look to avoid doing this. Is this necessary?

---

Do you mean that you'd like to remove the id prop?

---

If you meant to remove the id prop, I noticed it's a common practice throughout the codebase, so I’d suggest keeping it for now.